### ISSUE TITLE
Best command line settings for outputting w forced subs

### ISSUE THREAD

**Issue by mumblequatch** [lisamelton/video_transcoding#327]
Hi there,

I'm a super-novice Github/Terminal/script user — I've more or less never used the command line to interact with my Mac (though I'm curious to learn more). 

But I've installed other-transcode and managed to get it working, yielding a *very* nice output! 

However, I would like to add forced subs to the MKV file that I output and am confused about the way the embedded help file puts it. I would appreciate very much a simple copy/paste solution!

To clarify, I'm simply trying to output a compressed MKV file that would include forced subs. To take a random example: the Italian restaurant scene in The Godfather. In that scene, Sollozzo says to McCluskey that he's going to speak to Michael in Italian and they go back and forth between speaking English and Italian. There are no subtitles when they're speaking English (i.e. no closed captions) but there are subtitles when they speak Italian. This is what I'm trying to achieve. Another example would be whenever characters speak Elvish in the Lord of the Rings movies. I want the Elvish subtitled only, not the English dialogue in general. 

So what would the commands be to obtain this output?

Thanks very much in advance! I'm a filmmaker so I make content for a living and obsess constantly over image and sound quality and FWIW, I'm really impressed with the quality of this compression, especially in relation to the file size!

Thanks a bunch,

T

**Comment by donmelton** [comment]
@mumblequatch Thank you for using my tools.

The `transcode-video` tool will automatically burn any forced subtitle into video if that subtitle track has the "forced" flag set in the metadata of your input MKV file.

However, if the "forced" flag is not set in the metadata, you can explicitly select which subtitle is burned into video using the `--burn-subtitle` option with the track number as the argument. IOW, to burn the first subtitle track, add this to your command line:

```
--burn-subtitle 1
```

If you would rather keep the subtitle separate from the video but add it to your output and set the force flag for that track, you can use the `--force-subtitle` option. IOW, to force the first subtitle track, add this to your command line:

```
--force-subtitle 1
```

Does that help?

BTW, I think you might be mistaken about two examples you gave of forced subtitles. I just re-watched "The Godfather" the other night and scene you describe does not have forced subtitles, ie. when Sollozzo speaks to Michael. Unlike the earlier scene when Sollozzo speaks to Luca Brasi which does. Coppola said he intentionally left out the forced subtitles with Sollozo and Michael because the content of what they were saying wasn't important. As for "The Lord of the Rings," both the Theatrical and Extended Edition discs have the Elvish subtitles pre-burned into the video, i.e. they are not separate tracks.

But there are plenty of movies and TV shows with forced subtitles around. A good example is "The Avengers" when the Black Widow is being held captive and questioned by the Russian mobsters.

**Comment by mumblequatch** [comment]
Yes, wonderfully helpful! 

One additional question, then: is there a command to automatically select the forced subtitle but keep it separate, as opposed to burning it in?

Regarding the examples, you're half right! In bringing up the Sollozzo scene I was just grabbing for a scene that I knew went back and forth between two languages, one of which was the primary language of the film and the other of which was not. But of course there were no subtitles on that scene and it's all the better for it!

But regarding the subtitles for Elvish, strangely there exists some version out there in common circulation that does _not_ have burned in subtitles for that language. I know because I was trying to watch LOTR recently with my wife and our Blu-ray disc for _The Fellowship of the Ring_ was not working, so I sought out a pirated version on BitTorrent (figuring that having paid in the theater more than once and owning the Blu-Ray made this okay) and: lo and behold, no burned-in subtitles on the Elvish. I mean, I suppose there are probably folks out there who actually do speak Elvish, but it's an odd choice to allow viewers to opt out of having subtitles on multiple scenes (which they would surely do accidentally) that contain important drama and exposition. 

At any rate, thanks very much for the helpful clarifications and for all that you do. Having long hesitated over the annoying settings in Handbrake, I'm excited to try bringing in more of my favorites and really get my Plex library going!

**Comment by donmelton** [comment]
@mumblequatch That's what the `--force-subtitle` option does. It keeps the subtitle as a separate track and sets the "forced" flag for that track in the output.

Whether a subtitle is burned in or included as a separate track often differs in the mastering of a Blu-ray Disc or DVD for non-English, or at least non-American, markets. For example, the forced subtitles for the first three Bourne films are pre-burned for the U.S. version but are separate tracks in the European version.

**Comment by mumblequatch** [comment]
Gooootcha, thank you. 

If MKV and your tool are both required in order to get from a commercial disc, either DVD or Blu-Ray, to a final MKV file, is there a way of stringing them together so that the process could be initiated and then walked away from? Does Automator have functionality like this? Sorry if I'm being dense — feel free to tell me to shut up or simply point the way generally!

**Comment by donmelton** [comment]
@mumblequatch It's possible to automate the ripping and transcoding process. However, I would not recommend it for the ripping portion because disc mastering varies so much that it's difficult to guarantee you can select the correct title and set the track attributes. Many people have tried this but I don't know of a solution that works well.

**Comment by muesic** [comment]
Not sure if I should have started another issue, but my question is related so I'll tack it on here. I have many ripped DVDs containing the just the main movie (original DVD compression) and the full spread of original audio tracks and subtitles. I wish to compress everything I have using the great tools here so that I end up with a compressed movie, only the best audio track (typically DD 5.1) and only a subtitle track if it is needed. Now without any doubt, the biggest hassle I have is not knowing in advance if a movie has forced subtitles or not. HandBrake (as near as I can tell) can't match the video compression quality here, but it does have a wonderful option whereby it can do a first pass Foreign Audio Search for forced subtitles and if it finds them then only output those. Do these tools offer such a feature - perhaps even via ffmpeg?

**Comment by donmelton** [comment]
@muesic You can try `--force-subtitle scan` with `transcode-video` to see if that works for you. All of these behaviors are detailed in `--help`.

**Comment by parkerbrother1** [comment]
@muesic What I do is extract the DVD with makemkv and under the subtitle I want, add 'f' under mkvflags.  transcode-video will automatically burn that subtitle.  You can use makemkv or mkvtoolnix with an existing mkv file also.

**Comment by muesic** [comment]
@donmelton: Well I just discovered that maybe I should have started a separate thread. I'm using other-transcode (vs transcode-video) and that doesn't have the --force-subtitle scan. I'll have to look a little deeper if I want to call HandBrakeCLI in my scripts of ffmpeg. Do both of your tools generate pretty much the same video quality/size results?

@parkerbrother1: Thanks for the tip - hopefully I can do all that via makemkvcon because I'm scripting everything. My recollection is that makemkvcon is woefully behind the GUI (MakeMKV) so I doubt this is possible.

**Comment by donmelton** [comment]
@muesic No, the two scripts are completely different. The `transcode-video` tool is no longer under active development. You should try to use `other-transcode` since I spend more time supporting that. If you have a question about `other-transcode` please first look in `--help full` and then open an issue on that project if you need additional help.
